packages/pds/src/db/migrations/20230323T202553064Z-rework-seq.ts

```diff
@@ -0,0 +1,100 @@
+import { Kysely, sql } from 'kysely'
+import { Dialect } from '..'
+
+const repoSeqTable = 'repo_seq'
+const repoOpTable = 'repo_op'
+const repoSeqDidIndex = 'repo_seq_did_index'
+const repoSeqCommitIndex = 'repo_seq_commit_index'
+const repoSeqEventTypeIndex = 'repo_seq_event_type_index'
+const repoSeqSequencedAtIndex = 'repo_seq_sequenced_at_index'
+
+export async function up(db: Kysely<unknown>, dialect: Dialect): Promise<void> {
+  await db.schema.dropIndex(repoSeqCommitIndex).execute()
+  await db.schema.dropIndex(repoSeqDidIndex).execute()
+  await db.schema.dropTable(repoSeqTable).execute()
+  await db.schema.dropTable(repoOpTable).execute()
+
+  let builder = db.schema.createTable(repoSeqTable)
+  if (dialect === 'pg') {
+    builder = builder
+      .addColumn('seq', 'bigserial', (col) => col.primaryKey())
+      .addColumn('invalidatedBy', 'bigint')
+  } else {
+    builder = builder
+      .addColumn('seq', 'integer', (col) => col.autoIncrement().primaryKey())
+      .addColumn('invalidatedBy', 'integer')
+  }
+
+  const binaryDatatype = dialect === 'sqlite' ? sql`blob` : sql`bytea`
+  await builder
+    .addColumn('did', 'varchar', (col) => col.notNull())
+    .addColumn('eventType', 'varchar', (col) => col.notNull())
+    .addColumn('event', binaryDatatype, (col) => col.notNull())
+    .addColumn('sequencedAt', 'varchar', (col) => col.notNull())
+    .execute()
+
+  await db.schema
```

Comment:
Having trouble recalling which types of queries these indices support.  I kinda like the comments we've added in some of the other migrations documenting this, have found it useful!

Reply:
yup i'll add some comments 👍